go/store/nbs/byte_sink_test.go

```diff
@@ -75,6 +77,73 @@ func TestBufferedFileByteSink(t *testing.T) {
 	})
 }
 
+func TestBlockBufferByteSink(t *testing.T) {
+	t.Run("Quota", func(t *testing.T) {
+		t.Run("NoError", func(t *testing.T) {
+			q := NewUnlimitedMemQuotaProvider()
+			sink, err := NewBlockBufferByteSink(t.Context(), 4096, q)
+			require.NoError(t, err)
+			assert.Equal(t, uint64(4096), q.Usage())
+			_, err = sink.Write(make([]byte, 4096))
+			require.NoError(t, err)
+			assert.Equal(t, uint64(4096), q.Usage())
+			_, err = sink.Write(make([]byte, 4096))
+			require.NoError(t, err)
+			assert.Equal(t, uint64(8192), q.Usage())
+			_, err = sink.Write(make([]byte, 0))
+			require.NoError(t, err)
+			assert.Equal(t, uint64(8192), q.Usage())
+			_, err = sink.Write(make([]byte, 1))
+			require.NoError(t, err)
+			assert.Equal(t, uint64(12288), q.Usage())
+			_, err = sink.Write(make([]byte, 4095))
+			require.NoError(t, err)
+			assert.Equal(t, uint64(12288), q.Usage())
+			_, err = sink.Write(make([]byte, 65536))
+			require.NoError(t, err)
+			assert.Equal(t, uint64(77824), q.Usage())
+			var buf bytes.Buffer
+			err = sink.Flush(&buf)
+			require.NoError(t, err)
+			assert.Equal(t, uint64(77824), q.Usage())
+			assert.Equal(t, 77824, buf.Len())
+			sink.Close()
+			assert.Equal(t, uint64(0), q.Usage())
+		})
+		t.Run("Error", func(t *testing.T) {
+			t.Run("Init", func(t *testing.T) {
+				q := &errorQuota{NewUnlimitedMemQuotaProvider(), 0}
+				_, err := NewBlockBufferByteSink(t.Context(), 4096, q)
```

Comment:
Nice to differentiate this call from the one three lines down.
```suggestion
				_, err := NewBlockBufferByteSink(t.Context(), 1, q)
```